tests/test_datasets/test_test_datasets_utils.py

```diff
@@ -0,0 +1,74 @@
+import random
+import unittest
+
+import requests
+
+from nf_core.test_datasets.test_datasets_utils import (
+    MODULES_BRANCH_NAME,
+    GithubApiEndpoints,
+    create_download_url,
+    create_pretty_nf_path,
+    get_remote_branch_names,
+    get_remote_tree_for_branch,
+    list_files_by_branch,
+)
+
+
+class TestTestDatasetsUtils(unittest.TestCase):
+    """Class for components tests"""
+
+    def setUp(self):
+        self.gh_urls = GithubApiEndpoints(gh_orga="nf-core", gh_repo="test-datasets")
+
+    def test_modules_branch_exists(self):
+        url = self.gh_urls.get_remote_tree_url_for_branch(MODULES_BRANCH_NAME)
+        resp = requests.get(url)
+        self.assertTrue(resp.ok)
+        self.assertTrue(len(resp.json()) != 0)
+
+    def test_get_branch_names(self):
+        branch_names = get_remote_branch_names()
+        self.assertTrue(len(branch_names) != 0)
+
+    def test_get_remote_tree_for_branch(self):
+        # get_remote_tree_for_branch
+        file_list = get_remote_tree_for_branch("modules")
```

Comment:
It would be good to mock the API calls to speed up tests, see [example here](https://github.com/nf-core/tools/blob/main/tests/utils.py#L65).